docs/ff-concepts/adding-customization/common-examples.md

```diff
@@ -0,0 +1,284 @@
+    setState(() {
+      isPressed = !isPressed;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      style: ElevatedButton.styleFrom(
+        primary: isPressed
+            ? FlutterFlowTheme.of(context).primary // Primary color when pressed
+            : FlutterFlowTheme.of(context).secondaryBackground, // Default color
+        onPrimary: FlutterFlowTheme.of(context).secondaryText, // Text color
+      ),
+      onPressed: toggleButton,
+      child: Text(
+        widget.label,
+        style: FlutterFlowTheme.of(context).bodyText1, // Themed text style
+      ),
+    );
+  }
+}
+```
+
+
+### Manipulating AppState from Custom Code
+
+In FlutterFlow, you can access or update AppState directly from the Action Flow Editor. However, certain scenarios may require you to access or modify AppState within custom code for more control over the operation flow. The FFAppState class also provides additional helper functions to manipulate AppState variables. Let’s look at some examples: 
+
+:::tip[Imports]
+Ensure you import `import '../../flutter_flow/flutter_flow_util.dart';` when accessing `FFAppState` in custom code resources.
+:::
+
+- **Accessing AppState in Custom Code** 
+
+```js
+
+Future getCartItems() async {
+  // Retrieve the current cart items from AppState
+  final currentCartItems = FFAppState().cartItems;
+  print('Current Cart Items: $currentCartItems');
+}
+```
+
+- **Updating AppState Values in Custom Code**
+
+```js
+Future enableDarkMode() async {
+  // Enable dark mode in AppState
+  FFAppState().update(() {
+    FFAppState().enableDarkMode = true;
+  });
+  print('Dark mode enabled');
+}
+```
+
+- **Modifying a List in AppState Using Helper Functions**
+
+The `FFAppState` class offers a variety of helper functions to easily manage list variables in AppState. For a detailed overview of this generated class, check out **[this guide](../../generated-code/ff-app-state.md#managing-appstatelist)**. Here’s some examples of how to use these helper functions to modify an AppState list variable:
+
+```js
+Future addLocation(LatLng value) async {
+  // Add a new location to the LatLng list
+  FFAppState().addToLatLngList(value);
+}
+
+Future removeLocation(LatLng value) async {
+  // Remove a specific location from the LatLng list
+  FFAppState().removeFromLatLngList(value);
+}
+
+Future removeLocationAtIndex(int index) async {
+  // Remove a location at a specific index from the LatLng list
+  FFAppState().removeAtIndexFromLatLngList(index);
+}
+
+Future updateLocationAtIndex(int index, LatLng Function(LatLng) updateFn) async {
+  // Update a location at a specific index in the LatLng list
+  FFAppState().updateLatLngListAtIndex(index, updateFn);
+}
+
+Future insertLocationAtIndex(int index, LatLng value) async {
+  // Insert a new location at a specific index in the LatLng list
+  FFAppState().insertAtIndexInLatLngList(index, value);
+}
+```
+
+
+### Manipulate Custom Data Types
```

Comment:
I wouldnt say manipulating here, we're not manipulating the underlying class. I would say "leveraging" instead